src/gui/dialog/SettingsDialog.cpp

```diff
@@ -539,7 +538,7 @@ void SettingsDialog::save()
 	settings->setInputSystemTPCButtonEnabled(getCheckbox("cbInputSystemTPCButton"));
 	settings->setInputSystemDrawOutsideWindowEnabled(getCheckbox("cbInputSystemDrawOutsideWindow"));
 
-	int scrollbarHideType = SCROLLBAR_HIDE_NONE;
+	unsigned scrollbarHideType = SCROLLBAR_HIDE_NONE;
```

Comment:
Please use `guint` as type otherwise Windows compilers may complain.

Reply:
I mostly develop c++ applications for windows this ist Not a Problem. Unsigned ist a inbuild type and also a compiler keyword. Also all enums derive drin unsigned If not other specified. Using guint would be a risky choice, because it could cause a typecast.

Reply:
The underlying type is implementation specific in C++11. Thus this might be the case for the compiler you are using.
As we are using `guint` as type for all other enums it is better if you stick with that.

Reply:
But this enum uses unsigned as underlying Type. And it is only a problem if you mix types, as assigning a gtype to an c++ type specifier. Thats again a reason, why it IS better to use always auto. In the case of enums auto wont work. Therefore i used the underlying type of the enum: unsigned. Guint can be problematik in this specific context.

Reply:
If we want to use guint as underlying type we must derive all enums explicitely from guint.

Reply:
`unsigned` is another name of `unsigned int` ([see here](https://de.cppreference.com/w/cpp/language/types)). Glib will also define `guint` as `unsigned int` if possible ([see here](https://people.gnome.org/~ryanl/glib-docs/glib-Basic-Types.html#guint)) but will use another definition if the compiler does not know of the type `unsigned int` which some Windows compilers do for some reason...

So this is only for compatibility reasons and nothing else...

Reply:
Please don't discuss to much on such simple problem.
Our target should be to create a good note talking application, and not to discuss such simple details.

As in this specific case, where the value is uses as bitmask it makes absolutely **no difference if the value is signed or not**. And there are really bad compilers, e.g. Visual Studio (yes, the current versions are much better than the old ones), which have a lot of type issues.
Currently we are compiling with GCC on Windows, but as open source project we should write code which is as portable as possible, therefore @LittleHuba is right with portability.
@Febbe is also right, that an enum should be unsigned.

Because of this compatibility stuff it's really not bad to use standard types like int, uint_32_t (stdint.h) or gint, which is handled by GTK.

But I also checked "unsigned" is listed on the visual studio supported types: https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=vs-2019

However: please focus at the real problems!

Reply:
The problem I had with guint is, that it is not more portable, because an enums size is defined as at least as large as a "int" or "unsigned". If the compiler interprets the size differently, it will also interpret the size of an enum differently (as his int size). A "guint" is defined on each platform as uint_32_t, there will be a type difference and an implicit type cast. Which is bug prone.

Ok, in this case there will be no difference. But if we use a guint as type we should also derive the enum from it.

The best way would be something like this: 
```
std::make_unsigned_t<std::underlying_type_t<enum_T>> name = ...
```
But thats realy ugly :D.